unsupported option '-mwindows' for target 'aarch64-w64-mingw32'

When doing a native build of NSPR 4.35 on Windows on Arm using [LLVM 17.0.0 RC1](https://github.com/mstorsjo/llvm-mingw/releases/tag/20230730) I got the error `unsupported option '-mwindows' for target 'aarch64-w64-mingw32'`.

However when building for MinGW-w64 the `-mwindows` is normally supported (at least I know it is for GCC).

Full output:
```console
make -C config export
make[1]: Entering directory '/r/winlibs_llvm17rc1/11064.nspr/nspr-4.35/nspr/config'
/d/Prog/winlibs_llvm17rc1/llvm-mingw-20230730-ucrt-aarch64/bin/clang.exe -mwindows -o now.o -c     -Wno-int-conversion -pthread -g   -UNDEBUG -DDEBUG_brecht -DPACKAGE_NAME=\"\" -DPACKAGE_TARNAME=\"\" -DPACKAGE_VERSION=\"\" -DPACKAGE_STRING=\"\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DDEBUG=1 -DXP_PC=1 -DWIN32=1 -DWINNT=1 -D_ARM64_=1 -DHAVE_STRERROR=1 -DFORCE_PR_LOG   now.c
clang-17: error: unsupported option '-mwindows' for target 'aarch64-w64-mingw32'
make[1]: *** [../config/rules.mk:450: now.o] Error 1
make[1]: Leaving directory '/r/winlibs_llvm17rc1/11064.nspr/nspr-4.35/nspr/config'
make: *** [config/rules.mk:163: export] Error 2
```

--- a/clang/lib/Driver/ToolChains/MinGW.cpp
+++ b/clang/lib/Driver/ToolChains/MinGW.cpp
@@ -709,8 +709,11 @@ void toolchains::MinGW::addClangTargetOptions(
     }
   }
 
-  if (Arg *A = DriverArgs.getLastArgNoClaim(options::OPT_mthreads))
-    A->ignoreTargetSpecific();
+  for (auto Opt : {options::OPT_mthreads, options::OPT_mwindows,
+                   options::OPT_mconsole, options::OPT_mdll}) {
+    if (Arg *A = DriverArgs.getLastArgNoClaim(Opt))
+      A->ignoreTargetSpecific();
+  }
 }
 
 void toolchains::MinGW::AddClangCXXStdlibIncludeArgs(

--- a/clang/test/Driver/mingw-linker-options.c
+++ b/clang/test/Driver/mingw-linker-options.c
@@ -1,0 +1,10 @@
+// RUN: %clang --target=x86_64-windows-gnu -c -mwindows %s -### 2>&1 | FileCheck %s --check-prefix=WARNING
+// RUN: %clang --target=x86_64-windows-gnu -c -mconsole %s -### 2>&1 | FileCheck %s --check-prefix=WARNING
+// RUN: %clang --target=x86_64-windows-gnu -c -mdll %s -### 2>&1 | FileCheck %s --check-prefix=WARNING
+// RUN: %clang --target=x86_64-windows-gnu -c -mthreads %s -### 2>&1 | FileCheck %s --check-prefix=WARNING
+// RUN: not %clang --target=x86_64-windows-msvc -c -mwindows %s -### 2>&1 | FileCheck %s --check-prefix=ERROR
+// RUN: not %clang --target=x86_64-windows-msvc -c -mconsole %s -### 2>&1 | FileCheck %s --check-prefix=ERROR
+// RUN: not %clang --target=x86_64-windows-msvc -c -mdll %s -### 2>&1 | FileCheck %s --check-prefix=ERROR
+// RUN: not %clang --target=x86_64-windows-msvc -c -mthreads %s -### 2>&1 | FileCheck %s --check-prefix=ERROR
+// WARNING: warning: argument unused during compilation: '{{.*}}' [-Wunused-command-line-argument]
+// ERROR: error: unsupported option '{{.*}}' for target '{{.*}}'

